src/common/updating/http_client.cpp

```diff
@@ -0,0 +1,79 @@
+#include "pch.h"
+#include "http_client.h"
+
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Web.Http.Headers.h>
+#include <winrt/Windows.Storage.Streams.h>
+
+namespace http
+{
+    using namespace winrt::Windows::Web::Http;
+    namespace storage = winrt::Windows::Storage;
+
+    const wchar_t USER_AGENT[] = L"Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.2; WOW64; Trident/6.0)";
+
+    HttpClient::HttpClient()
+    {
+        auto headers = client.DefaultRequestHeaders();
+        headers.UserAgent().TryParseAdd(USER_AGENT);
+    }
+
+    std::future<std::wstring> HttpClient::request(const winrt::Windows::Foundation::Uri& url)
+    {
+        auto response = co_await client.GetAsync(url);
+        (void)response.EnsureSuccessStatusCode();
+        auto body = co_await response.Content().ReadAsStringAsync();
+        co_return std::wstring(body);
+    }
+
+    std::future<void> HttpClient::download(const winrt::Windows::Foundation::Uri& url, const std::wstring& dstFilePath)
+    {
+        auto response = co_await client.GetAsync(url);
+        (void)response.EnsureSuccessStatusCode();
+        auto msi_installer_file_stream = co_await storage::Streams::FileRandomAccessStream::OpenAsync(dstFilePath.c_str(), storage::FileAccessMode::ReadWrite, storage::StorageOpenOptions::AllowReadersAndWriters, storage::Streams::FileOpenDisposition::CreateAlways);

```

Comment:
`msi_installer_file_stream` -> `file_stream` (perhaps it's due to conflicts with master)